### ISSUE TITLE
Update rule `aria-required-children`

### ISSUE THREAD

**Issue by jkodu** [dequelabs/axe-core#2076]
Expectation: 
For examples below, the rule should fail because required children role is not correct.

```
<div role="list">
	<div role="tabpanel">
		<div role="listitem">List item 1</div>
	</div>
</div>
```

```
<div role="list" aria-owns="item"></div>

<div id="host"></div>

<script>
	const host = document.querySelector('#host')
	const root = host.attachShadow({ mode: 'open' })
	root.innerHTML = '<div id="item" role="listitem">List item 1</div>'
</script>
```

Actual:
The rule only looks for children with an allowed role, without intercepting nodes in between which can have over-ridding roles.

Motivation:
Required for implementation of rule in ACT Rule - https://act-rules.github.io/rules/bc4a75

**Comment by padmavemulapati** [comment]
Verified with latest axe-core, and with the above specified test example, aria-required-children rule is failing , which is expected.
![image](https://user-images.githubusercontent.com/47409579/78759799-d2b31700-799d-11ea-9bfd-0bb8dbade1e8.png)

**Comment by straker** [comment]
No docs
